src/main/java/greencity/repository/options/UserFilter.java

```diff
@@ -52,11 +53,29 @@ public Predicate toPredicate(Root<User> root, CriteriaQuery<?> criteriaQuery, Cr
      * @return a {@link Predicate}, may be {@literal null}.
      */
     private Predicate hasFieldsLike(Root<User> r, CriteriaBuilder cb, String reg) {
+        reg = replaceCriteria(reg);
         return cb.or(
             cb.like(r.get(RepoConstants.FIRST_NAME), reg),
             cb.like(r.get(RepoConstants.LAST_NAME), reg),
             cb.like(r.get(RepoConstants.EMAIL), reg),
             cb.like(r.get(RepoConstants.REGISTRATION_DATE).as(String.class), reg)
         );
     }
+
+    /**
+     * Returns a String criteria for search.
+     *
+     * @param criteria  must not be {@literal null}.
+     * @return String creteria  may be {@literal null}.
+     */
+    private String replaceCriteria(String criteria) {
+        criteria = Optional.of(criteria).orElse("");
```

Comment:
For what purpose do you use this assignment?
If criteria will have null value, you get NullPointerException. So you should check null value.

Reply:
Check again please.

Reply:
* @param criteria  must not be {@literal null}.
      * @return String creteria  may be {@literal null}
If criteria must not be null you shoud throw exception. If criteria can be null then you should change description.
And how can you return null if you assign criteria to a new value, when you get null as an argument?

Reply:
If you want to use null check you can use criteria=Optional.ofNullable(criteria).orElse("");
Optional.of throw exception when argument null